src/Controls/src/Core/ResourceDictionary.cs

```diff
@@ -47,7 +47,7 @@ public void SetAndLoadSource(Uri value, string resourcePath, Assembly assembly,
 			//this will return a type if the RD as an x:Class element, and codebehind
 			var type = XamlResourceIdAttribute.GetTypeForPath(assembly, resourcePath);
 			if (type != null)
-				_mergedInstance = s_instances.GetValue(type, (key) => (ResourceDictionary)Activator.CreateInstance(key));
+				_mergedInstance = s_instances.GetValue(type, _ => (ResourceDictionary)Activator.CreateInstance(type));
```

Comment:
This creates a closure on `type` where there wasn't before, and would allocate. Could a new `static` method with appropriate attributes avoid this?

Reply:
We could do this with a suppression - not pretty.
Is this a hot path where allocations matter a lot?

Reply:
Looking at the project template, I think this would only get called twice on startup:

https://github.com/dotnet/maui/blob/71b7316c915aaeb558a1b060a2616cc6a036cd92/src/Templates/src/templates/maui-mobile/App.xaml#L9-L10

So if the code would be quite ugly, we can keep this as-is.